Sources/DeepLearning/Operators.swift

```diff
@@ -120,91 +120,78 @@ public extension Padding {
         strides: (Int32, Int32, Int32, Int32),
         padding: Padding
     ) -> Tensor {
         return Raw.conv2DBackpropFilter(
             input,
             filterSizes: filterSizes,
-            outBackprop: backpropOutput,
+            outBackprop: self,
             strides: [strides.0, strides.1, strides.2, strides.3],
             padding: padding.raw)
     }
-
+    
     @inlinable
-    func _vjpTFConv2DBackpropInput(
+    func _vjpConv2DBackpropInput(
         _ shape: Tensor<Int32>,
         _ filter: Tensor,
-        _ backpropOutput: Tensor,
         _ strides: (Int32, Int32, Int32, Int32),
         _ padding: Padding
     ) -> (Tensor, (Tensor) -> (Tensor, Tensor)) {
-        let value = _TFConv2DBackpropInput(shape: shape, filter: filter,
-                                           backpropOutput: backpropOutput,
-                                           strides: strides, padding: padding)
+        let value = conv2DBackpropInput(shape: shape, filter: filter, strides: strides,
+                                             padding: padding)
         return (value, { v in
             return (
-                self._TFConv2DBackpropFilter(input: v, filterSizes: shape,
-                                             backpropOutput: backpropOutput,
-                                             strides: strides, padding: padding),
+                self.conv2DBackpropFilter(input: v, filterSizes: shape, strides: strides,
+                                               padding: padding),
                 v.convolved2D(withFilter: filter, strides: strides, padding: padding)
             )
         })
     }
-
+    
     @inlinable
-    func _vjpTFConv2DBackpropFilter(
+    func _vjpConv2DBackpropFilter(
         _ input: Tensor,
         _ filterSizes: Tensor<Int32>,
-        _ backpropOutput: Tensor,
         _ strides: (Int32, Int32, Int32, Int32),
         _ padding: Padding
     ) -> (Tensor, (Tensor) -> (Tensor, Tensor)) {
-        let value = _TFConv2DBackpropFilter(input: input, filterSizes: filterSizes,
-                                            backpropOutput: backpropOutput,
-                                            strides: strides, padding: padding)
+        let value = conv2DBackpropFilter(input: input, filterSizes: filterSizes,
+                                              strides: strides, padding: padding)
         return (value, { v in
             return (
-                self._TFConv2DBackpropInput(shape: filterSizes, filter: v,
-                                            backpropOutput: backpropOutput,
-                                            strides: strides, padding: padding),
+                self.conv2DBackpropInput(shape: filterSizes, filter: v, strides: strides,
+                                              padding: padding),
                 input.convolved2D(withFilter: v, strides: strides, padding: padding)
             )
         })
     }
-
+    
     @inlinable
     func _vjpConvolved2D(
```

Comment:
The extension is public, but all VJP functions should be explicitly `internal`.

Reply:
Done.